pixeltable/dataframe.py

```diff
@@ -454,3 +465,54 @@ def __getitem__(self, index: object) -> DataFrame:
         if isinstance(index, list):
             return self.select(*index)
         raise TypeError(f'Invalid index type: {type(index)}')
+    
+    def as_dict(self) -> Dict[str, Any]:
+        """ 
+            Returns:
+                Dictionary representing this dataframe.
+        """
+        assert self.group_by_clause == [] or self.group_by_clause is None, 'non trivial group_by not yet supported in this method'
+        assert self.order_by_clause == [] or self.order_by_clause is None, 'non trivial order_by not yet supported in this method'
+
+        d = {
+            'type': 'DataFrame',
+            'tbl_id': str(self.tbl.id),
+            'tbl_version': self.tbl.version,
+            'select_list': [(e.as_dict(), name) for (e, name) in self.select_list] if self.select_list is not None else None,
+            'where_clause': self.where_clause.as_dict() if self.where_clause is not None else None,
+            'group_by_clause': None,
+            'order_by_clause': None,
+            'column_types': {name: col_type.as_dict() for (name, col_type) in zip(self.get_column_names(), self.get_column_types())}
+        }
+        return json.loads(json.dumps(d)) # check that it is serializable
+        
+    def to_pytorch_dataset(self, image_format : str = 'pt') -> 'torch.utils.data.IterableDataset':
+        """
+            Convert the dataframe to a pytorch IterableDataset suitable for parallel loading
+             with torch.utils.data.DataLoader. 
+
+            This method serializes data so it can be read from disk efficiently, repeatedly without 
+            re-executing the query. This data is cached to disk for future re-use.
+
+            Args:
+                image_format: format of the images. Can be 'pt' (pytorch tensor) or 'np' (numpy array).
+                        'np' means image columns return as an RGB uint8 array of shape HxWxC.
```

Comment:
While Torch dataloader output batches are generally torch tensors, Torch dataset is a general class, it is not uncommon to get numpy stuff out of them, and torch dataloader  does not assume everything coming in to it is a torch tensor: It handles numpy just as well. (more accurately, this is part of what the collate_fn argument does), so this gives us some flexibility. In terms of why I return them: 

For image types,  numpy rep of images can be easily converted to PIL.Image, allowing  easy inspection of the data.  This is one big reason to keep them.  Pytorch image tensors aren't just different  in type, they have their own conventions on channel ordering and scaling.  This pytorch representation is ok when passing it to a network, but not good if you want to view the image or if you wanted something non-standard. 

For array types, it makes sense to pass np array because that is our native image representation, and pyarrow also understands this format: If we convert to pytorch ourselves then we need to write tests checking the right conversions have happened, and that we haven't lost information.  So making everything pytorch seems to add complexity, but for little in return, because most downstream classes including torch dataloader are ok with numpy arrays and handle conversion on their own. By keeping them in numpy neither us nor users need to understand implicit conversions from our end: the values should be identical to what they saved.